### DISCUSSION TITLE
Make delay by buffers visible in Settings-UI

### DISCUSSION THREAD

**Discussion by nefarius2001** [jamulussoftware/jamulus#1257] [Ideas]
Setting up a system, buffers and settings are tweaked.
There was no easy way to see how many ms delay come from the buffers, only overall and ping time were given. This was especially of interrest to comapre the "fixed" buffers to "variable" network delays.
I propose an additional line showing the delay from network & audio buffers. The label should show the simple number, while tooltip of the label can show the details.

I wrote this for myself already (https://github.com/nefarius2001/jamulus/tree/feature/delaydetails), but am happy to implement changes/improvements.

(the example pictures have german language)
![grafik](https://user-images.githubusercontent.com/6545169/103765098-e79c0400-501c-11eb-9faa-7acc4e7d9ab3.png)
![grafik](https://user-images.githubusercontent.com/6545169/103764987-be7b7380-501c-11eb-80f5-a73b40a6c957.png)

**Comment by ann0see** [comment]
I like your proposal. It might show issues with the buffer/network (?) settings.

However, I think we should try not to over-complicate this page (don't try to show too much data). Could or should we move the "Buffer Delay" to the tool tip and show it on hovering "Gesamtverzögerung" only?

**Comment by nefarius2001** [comment]
That is possible, since it is kind-of static information ( a normal tooltip can not easily be updated live while it is shown).

I thought about the complexity, too. In my opinion the "overall delay" is the "top-level" statistic, and only that one should be visible by default. The pure ping-time and the buffer-delay are kind of second-level. I would prefer having those in some kind of expand-tree view (if that exists). I prefer that to be acessible right away on changing it, not change-hover-wait-read-change-hover-...

What do you think?

**Comment by ann0see** [comment]
> I would prefer having those in some kind of expand-tree view 

Yes. That would be a good option.

**Comment by gilgongo** [comment]
In the interests of getting a resolution on this, moving to discussion.
